core/field_multilineinput.js

```diff
@@ -211,6 +228,27 @@ Blockly.FieldMultilineInput.prototype.updateSize_ = function() {
     totalHeight += this.getConstants().FIELD_TEXT_HEIGHT +
         (i > 0 ? this.getConstants().FIELD_BORDER_RECT_Y_PADDING : 0);
   }
+  if (this.isBeingEdited_) {
```

Comment:
As you type in the editor, the underlying SVG text changes, and the width of the block is originally set as the longest SVG text element (above this section).     
When we restrict the number of lines, then the block width is always restricted by the first `maxLines` number of SVG lines.    
This results in a behavior where if `maxLines` is exceeded in editor mode, editor size no longer matches the longest line in the editor.       
So here I set the block width as the longest string in editor, if it is being edited. This is important, otherwise you will have situations where the first `maxLines` lines have length of say 1, and then if another `maxLines+1`-th line followed with more characters, it would keep breaking down into tiny peaces, as shown below.

here's what it looks like without this section. The last line should not break but should resize the editor.
![before2](https://user-images.githubusercontent.com/42584490/100829177-cc643480-3471-11eb-853a-51cd9d1885b5.gif)
With this section enabled:
![after2](https://user-images.githubusercontent.com/42584490/100829300-13522a00-3472-11eb-9f56-adfc27fda4bc.gif)

Reply:
Also, the difference you were talking about here, that you had noticed, seems to be it was a bug, I think it's fixed now. That was happening because I was scaling the fontsize here before calculating line width. Actually, I still don't know why, but it seems to work perfectly fine without that, and even the bug is fixed :)

Reply:
Thanks, makes sense, and glad you were able to fix the resizing issue! Could you add a comment summarizing what you said here? Something like "The default width is based on the longest line in the display text, but we need to calculate based on the absolute longest line even if it would be truncated after editing."